File: TeeNew/Builder2006/Surface_Sides.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Surface_Sides.h"
#include <TeeBrushDlg.hpp>
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeSurfa"
#pragma resource "*.dfm"
TSurfaceSidesDemo *SurfaceSides;
//---------------------------------------------------------------------------
__fastcall TSurfaceSidesDemo::TSurfaceSidesDemo(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TSurfaceSidesDemo::CheckBox1Click(TObject *Sender)
{
  if (CheckBox1->Checked)
  {
    Series1->SideBrush->Style = bsDiagCross;
    Series1->SideBrush->Color = clBlue;
  }
  else Series1->SideBrush->Style = bsClear;
}
//---------------------------------------------------------------------------

void __fastcall TSurfaceSidesDemo::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(12);
}
//---------------------------------------------------------------------------

void __fastcall TSurfaceSidesDemo::Button1Click(TObject *Sender)
{
  EditChartBrush(this,Series1->SideBrush);
}
//---------------------------------------------------------------------------

File: TeeNew/Surface_Sides.pas
unit Surface_Sides;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, Series, TeeSurfa, TeeGDIPlus {, TeeGDIPlus};

type
  TSurfaceSides = class(TBaseForm)
    Series1: TSurfaceSeries;
    CheckBox1: TCheckBox;
    Button1: TButton;
    procedure CheckBox1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Button1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeBrushDlg;

procedure TSurfaceSides.CheckBox1Click(Sender: TObject);
begin
  inherited;
  if CheckBox1.Checked then
  begin
    Series1.SideBrush.Style:=bsDiagCross;
    Series1.SideBrush.Color:=clBlue;
  end
  else Series1.SideBrush.Style:=bsClear;
end;

procedure TSurfaceSides.FormCreate(Sender: TObject);
begin
  inherited;
  Chart1.View3D := True;
  Series1.FillSampleValues(12);
end;

procedure TSurfaceSides.Button1Click(Sender: TObject);
begin
  EditChartBrush(Self,Series1.SideBrush);
end;

initialization
  RegisterClass(TSurfaceSides);
end.
